plugin/arcconf/arcconf.py

```diff
@@ -0,0 +1,919 @@
+                else:
+                    continue
+
+        if component == 'Logical Device':
+            for info_section in getconfig_logical_info_section:
+                if info_section == temp_list[0]:
+                    logical_data.update({temp_list[0] : temp_list[1]})
+                elif temp_list[0].find('Segment ') != -1:
+                    serialNo = temp_list[1].split(')')[1]
+                    serialNo = serialNo.strip()
+                    logical_data.update({'Serial number' : serialNo})
+                    list_data.append(logical_data.copy())
+                    break
+                else:
+                    continue    
+
+    if component == "Adapter":
+        return cntrl_data
+    return list_data
+
+
+def _arcconf_size_to_lsm(arcconf_size):
+    """
+    ARCCONF displays the disk size in terms of  'TB, GB, MB, KB'
+    for LSM, they are 'TiB' and etc.
+    Returning int of block bytes
+    """
+    re_regex = re.compile("^([0-9.]+) +([EPTGMK])B")
+    re_match = re_regex.match(arcconf_size)
+    if re_match:
+        return size_human_2_size_bytes(
+            "%s%siB" % (re_match.group(1), re_match.group(2)))
+
+    raise LsmError(
+        ErrorNumber.PLUGIN_BUG,
+        "_arcconf_size_to_lsm(): Got unexpected size string %s" %
+        arcconf_size)
+
+def _lsm_size_to_arcconf_MB(lsm_size):
+    """
+    Arcconf normally follows the size to be in MB
+    """
+    return ((lsm_size/1024)/1024)
+
+
+def _pool_status_of(arcconf_array):
+    """
+    Return (status, status_info)
+    """
+    if arcconf_array[0]['Status'] == 'OK':
+        return Pool.STATUS_OK, ''
+    elif arcconf_array[0]['Status'] == 'Optimal':
+        return Pool.STATUS_OK, ''
+    elif arcconf_array[0]['Status'] == 'Ready':
+        return Pool.STATUS_OK, ''
+    elif arcconf_array[0]['Status'] == 'Online':
+        return Pool.STATUS_OK, ''
+    else:
+        # TODO(Gris Ge): Try degrade a RAID or fail a RAID.
+        return Pool.STATUS_OTHER, arcconf_array['Status']
+
+
+def _pool_id_of(sys_id, array_name):
+    return "%s:%s" % (sys_id, array_name.replace(' ', ''))
+
+
+def _disk_type_of(arcconf_disk):
+    disk_interface = arcconf_disk['interfaceType']
+    isSSD = arcconf_disk['nonSpinning']
+
+    if isSSD == True:
+        return Disk.TYPE_SSD 
+
+    if disk_interface == 1:
+        return Disk.TYPE_SATA
+    elif disk_interface == 4:
+        return Disk.TYPE_SAS
+
+    return Disk.TYPE_UNKNOWN
+
+
+def _disk_status_of(arcconf_disk, flag_free):
+    # TODO(Gris Ge): Need more document or test for non-OK disks.
+    state = arcconf_disk['state']
+
+    if state == DRIVE_READY or state == DRIVE_ONLINE:
+        disk_status = Disk.STATUS_OK
+    else:
+        disk_status = Disk.STATUS_OTHER
+
+    if flag_free:
+        disk_status |= Disk.STATUS_FREE
+
+    return disk_status
+
+
+def _disk_link_type_of(arcconf_disk):
+    disk_interface = arcconf_disk['interfaceType']
+    if disk_interface == 1:
```

Comment:
Please don't use magic number.
Please try to remove the duplicate code with `_disk_type_of()`.